Shopipy.Web/Services/UserService.cs

```diff
@@ -0,0 +1,26 @@
+using Shopipy.UserManagement.Dtos;
+using System.Net.Http.Headers;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Shopipy.Web.Services;
+
+public class UserService(IHttpClientFactory httpClientFactory)
+{
+    private readonly HttpClient _httpClient = httpClientFactory.CreateClient("Shopipy.ApiService");
+
+    public async Task<UserResponseDto?> GetUserByIdAsync(string userId, string token)
```

Comment:
Put user info caching here

Reply:
caching in to what

Reply:
Sorry, meant caching for current user (not all the possible calls). You could have a method for that.